Update GoDoc for Azure.

negz opened the pull request:
<!--
Thank you for helping to improve Crossplane!

We strongly recommend you look through our contributor guide at https://git.io/fj2m9
if this is your first time opening a Crossplane pull request. You can find us in
https://slack.crossplane.io/messages/dev if you need any help contributing.
-->

### Description of your changes
<!--
Briefly describe what this pull request does. Be sure to direct your reviewers'
attention to anything that needs special consideration.

We love pull requests that resolve an open Crossplane issue. If yours does, you
can uncomment the below line to indicate which issue your PR fixes, for example
"Fixes #500":

Fixes #
-->
This documentation is used to generate our user facing documentation, both in CRDs and on the website. I've edited it for tone, and to generally make sense where it did not.

### Checklist
<!--
Please run through the below readiness checklist. The first two items are
relevant to every Crossplane pull request.
-->
I have:
- [ ] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Ensured this PR contains a neat, self documenting set of commits.
- [ ] Updated any relevant [documentation], [examples], or [release notes].
- [ ] Updated the RBAC permissions in [`clusterrole.yaml`] to include any new types.

[documentation]: https://github.com/crossplaneio/crossplane/tree/master/docs
[examples]: https://github.com/crossplaneio/crossplane/tree/master/cluster/examples
[release notes]: https://github.com/crossplaneio/crossplane/tree/master/PendingReleaseNotes.md
[`clusterrole.yaml`]: https://github.com/crossplaneio/crossplane/blob/master/cluster/charts/crossplane/templates/clusterrole.yaml

hasheddan commented on azure/apis/cache/v1alpha2/redis_types.go:
```diff
@@ -122,15 +122,21 @@ type SKUSpec struct {
 	runtimev1alpha1.ResourceStatus `json:",inline"`
 
-	State   string `json:"state,omitempty"`
-	Message string `json:"message,omitempty"`
```
Is the `Message` field supposed to be completely removed?

negz replied:
Yeah that was intentional - it's never actually set anywhere. I suspect this is the case for a lot of our resources, but this one I checked.

hasheddan commented on azure/apis/compute/v1alpha2/types.go:
```diff
@@ -25,84 +25,104 @@ import (
-	// ResourceGroupName is the name of the resource group that the cluster will be created in
-	ResourceGroupName string `json:"resourceGroupName"` //--resource-group
+	// ResourceGroupName is the name of the resource group that the cluster will
+	// be created in
```
I notice a number of these comments do not end with a period. Is that intentional?

negz replied:
I'd prefer them to end with periods, but I'm sure I missed them in a bunch of places since I'm trying to power through all this pretty quickly.

hasheddan replied:
Sounds good! I'll try to go through and catch some, probably after v0.3 :)

hasheddan commented on azure/apis/network/v1alpha2/types.go:
```diff
@@ -1,17 +1,16 @@
-You may obtain a copy of the License at
+Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+this file except in compliance with the License. You may obtain a copy of the
+License at
```
Looks like license header was changed here but not elsewhere, is that intended?

negz replied:
Ah I think this was me trying out that reformat extension you recommended. Will fix.

hasheddan commented on azure/apis/storage/v1alpha2/account.go:
```diff
@@ -379,7 +384,7 @@ func toStorageSkuCapability(s skuCapability) storage.SKUCapability {
 }
 
-// Sku the Sku of the storage account.
+// Sku of an Azure Blob Storage Account.
```
I apologize on behalf of Azure lol

negz commented:
@hasheddan PTAL - I believe I addressed all the comments except for the missing periods, which I think we can fix up as we spot them (or during code generation).

Speaking of code generation, one interesting observation is that many of the Azure SDK types seem to be a little lacking in the comment department, with comments like `// Foo - Gets or sets the Foo`, or just `// Foo..`, so it may not be a given that we can get high quality documentation of our fields simply by inspecting the SDK types. :(
